File: Check Automorphic Number/check_automorphic_number.c
#include <stdio.h>
int main() {
    int n, square, temp1, temp2;
    printf("Enter a number: ");
    scanf("%d", &n);
    square = n * n;
    temp1 = n;
    temp2 = square;
    int digitsN = 0;
    while (temp1 > 0) {
        temp1 = temp1 / 10;
        digitsN++;
    }
    int divisor = 1;
    for (int i = 0; i < digitsN; i++) {
        divisor = divisor * 10;
    }
    if (square % divisor == n) {
        printf("%d is an automorphic number\n", n);
    } else {
        printf("%d is not an automorphic number\n", n);
    }
    return 0;
}

File: Check Automorphic Number/CheckAutomorphicNumber.java
public class CheckAutomorphicNumber {
    public static void main(String[] args) {
        java.util.Scanner sc = new java.util.Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = sc.nextInt();
        int square = n * n;
        String nStr = String.valueOf(n);
        String squareStr = String.valueOf(square);
        if (squareStr.endsWith(nStr)) {
            System.out.println(n + " is an automorphic number");
        } else {
            System.out.println(n + " is not an automorphic number");
        }
    }
}
